modules/provision/debug.go

```diff
@@ -0,0 +1,116 @@
+package provision
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zosv2/modules/stubs"
+	"github.com/threefoldtech/zosv2/modules/zinit"
+
+	"github.com/pkg/errors"
+)
+
+// Debug provision schema
+type Debug struct {
+	Host    string `json:"host"`
+	Port    int    `json:"port"`
+	Channel string `json:"channel"`
+}
+
+func debugProvision(ctx context.Context, reservation *Reservation) (interface{}, error) {
+	var cfg Debug
+	if err := json.Unmarshal(reservation.Data, &cfg); err != nil {
+		return nil, err
+	}
+
+	return startZLF(ctx, reservation.ID, cfg)
+}
+
+func debugDecommission(ctx context.Context, reservation *Reservation) error {
+	return stopZLS(ctx, reservation.ID)
+}
+
+func startZLF(ctx context.Context, ID string, cfg Debug) (string, error) {
+	zbus := GetZBus(ctx)
+	identity := stubs.NewIdentityManagerStub(zbus)
+
+	path, err := exec.LookPath("zlf")
+	if err != nil {
+		return "", errors.Wrap(err, "failed to start zlf")
+	}
+
+	z := zinit.New("")
+	if err := z.Connect(); err != nil {
+		return "", errors.Wrap(err, "fail to connect to zinit")
+	}
+	defer z.Close()
+
+	channel := fmt.Sprintf("%s-logs", identity.NodeID().Identity())
+	if cfg.Channel != "" {
+		channel = cfg.Channel
+	}
+
+	s := zinit.InitService{
+		Exec:    fmt.Sprintf("%s --host %s --port %d --channel %s", path, cfg.Host, cfg.Port, channel),
+		Oneshot: false,
+		After:   []string{"networkd"},
+	}
+
+	name := fmt.Sprintf("zlf-debug-%s", ID)
+	if err := zinit.AddService(name, s); err != nil {
+		return "", errors.Wrap(err, "fail to add init service to zinit")
+	}
+
+	if err := z.Monitor(name); err != nil {
+		return "", errors.Wrap(err, "failed to start monitoring zlf service")
+	}
+
+	if err := z.Start(name); err != nil {
```

Comment:
You don't need to call start, monitor will auto start it.
you only need to call start if you previously did a `stop`

Reply:
got it